src/main/resources/db/migration/V1_117__update_print_files_to_clob_procedure.sql

```diff
@@ -0,0 +1,168 @@
+-- DROP PROCEDURE juror_mod.printfiles_to_clob(int4);
+
+CREATE OR REPLACE PROCEDURE juror_mod.printfiles_to_clob(IN p_limit integer DEFAULT 1000000)
+ LANGUAGE plpgsql
+AS $procedure$
+DECLARE
+	v_form_type VARCHAR(6);
+	v_max_rec_len INTEGER;
+	v_ext_date DATE;
+	forms RECORD;
+	v_count INTEGER:=0;
+
+BEGIN
+	v_ext_date :=	CASE
+						WHEN TO_CHAR(NOW(),'SSSSS')::INTEGER < 64800
+              				THEN NOW()::date - 1
+               				ELSE NOW()
+              		END;
+
+    -- delete redundant letter requests
+	CALL juror_mod.delete_printfiles();
+
+	FOR forms IN
+		SELECT	fr.form_type,
+				fr.max_rec_len::integer
+		FROM juror_mod.t_form_attr fr
+
+		loop
+			CALL juror_mod.write_printfiles(forms.form_type, forms.max_rec_len, v_ext_date, p_limit, v_count);
+			IF v_count >= p_limit THEN
+				EXIT;
```

Comment:
See comment in https://centralgovernmentcgi.atlassian.net/browse/JM-7397 about The maximum number of records per file not being handled as Heritage and check with Liam regarding the question put to him in that comment about asking the client whether the limit is needed.